src/client/driver/driver.js

```diff
@@ -426,6 +444,139 @@ export default class Driver extends serviceUtils.EventEmitter {
         childIframeDriverLink.sendConfirmationMessage(id);
     }
 
+    _getTargetWindowNotFoundResult () {
+        return Promise.resolve({
+            success: false,
+            errCode: TEST_RUN_ERRORS.targetWindowNotFoundError
+        });
+    }
+
+    _getChildWindowValidateResult (arr) {
+        const success = arr.find(item => item.result.success);
+
+        if (success)
+            return success.result;
+
+        let errItem = arr.find(item => item.result.errCode === TEST_RUN_ERRORS.cannotCloseWindowWithChildrenError);
+
+        if (!errItem)
+            errItem = arr.find(item => !!item.result.errCode);
+
+        return errItem ? { errCode: errItem.result.errCode } : void 0;
+    }
+
+    _handleWindowValidation (msg, wnd, windowFoundHandler, WindowValidationMessageCtor) {
+        let promiseResult = null;
+
+        if (msg.windowId === this.windowId)
+            promiseResult = windowFoundHandler();
+
+        else if (!this.childWindowDriverLinks.length) {
+            promiseResult = Promise.resolve({
+                success: false,
+                errCode: TEST_RUN_ERRORS.targetWindowNotFoundError
+            });
+        }
+
+        else {
+            promiseResult = Promise.all(this.childWindowDriverLinks.map(childWindowDriverLink => {
+                return childWindowDriverLink.searchChildWindows(msg, WindowValidationMessageCtor);
+            }))
+                .then(arr => {
+                    return this._getChildWindowValidateResult(arr);
+                });
+        }
+
+        return promiseResult
+            .then(result => {
+                sendConfirmationMessage({
+                    requestMsgId: msg.id,
+                    window:       wnd,
+                    result
+                });
+            });
+    }
+
+    _handleCloseWindowValidation (msg, wnd) {
+        const windowFoundHandler = () => {
```

Comment:
I think if we could refactor `windowFoundHandler` to a method and use an arrow `() => this.windowFoundHandler()` in place to improve readability.